### PR TITLE
adding api key expectation at the cdn layer

### PR DESCRIPTION
**Pull request by wheresrhys**
@Financial-Times/next-myft

### CODE REVIEW

File: vcl/main.vcl [hunk 1]
```diff
@@ -179,56 +180,16 @@ sub vcl_deliver {
...
 	if (obj.status == 755) {
-		set obj.status = 200;
-		synthetic {"FASTLY OK"};
+		set obj.status = 401;
+		set obj.http.Content-Type = "application/json; charset=utf-8";
+		synthetic {"{'status': 401,'message': 'Unauthorised: please supply valid API key.'}"};
 		return(deliver);
 	}
 
-	if (obj.status == 801) {
```
Comment (phamann) [inline vcl/main.vcl]: Why have you removed this? It is still needed, as we are forcing everyone onto HTTPs.

File: vcl/main.vcl [hunk 2]
```diff
@@ -46,9 +50,6 @@ sub vcl_recv {
 		set req.http.Protocol = "http://";
 	}
 
-	set req.http.FT-Next-Backend-Key = "${FT_NEXT_BACKEND_KEY}";
```
Comment (phamann) [inline vcl/main.vcl]: This is still needed.

Comment (wheresrhys) [inline vcl/main.vcl]: It's not really. Only intended for apps for whom the only thing stopping them being public is the fact the next.ft.com CDN sets this. We should call next-express with `withBackendAuthentication: false` to opt out.

File: vcl/main.vcl (hunk 1 above)
Comment (wheresrhys) [inline vcl/main.vcl]: ah yes - will put it back

Comment (wheresrhys) [inline vcl/main.vcl]: but the fake html error page below it not necessary

**Comment by wheresrhys** [comment]
see https://myft-api-staging.ft.com/v1/user/3f041222-22b9-4098-b4a6-7967e48fe4f7/followed https://myft-api.ft.com/v1/user/3f041222-22b9-4098-b4a6-7967e48fe4f7/followed

**Pull request merged**